gamemode/load_modules.lua

```diff
@@ -2,23 +2,52 @@ local moduleDir = GM.FolderName .. "/gamemode/modules/"
 
 local _, modules = file.Find( moduleDir .. "*", "LUA" )
 
+print( "[RVR] Loading modules..." )
 for _, moduleName in pairs( modules ) do
     local path = moduleDir .. moduleName .. "/"
     local files, _ = file.Find( path .. "*.lua", "LUA" )
-    
-    for _, file in pairs( files ) do
-        local fullPath = path .. file
+    local isServer = false
+    local isClient = false
```

Comment:
Maybe we could initialize these as `true` and make them false when it doesn't apply? I'm thinking it could save two lines since a module is never not server-side and not client side

Reply:
Don't think that logic would work, say a module had a shared + client file, it'd set `isServer` to false in the client file, and see the module as client instead of shared.

Reply:
Oh I see... Could we break up the modules into their own different prints then? I think it will clear up what actually loaded if there were ever the case of having a shared, client, and server module all broken into different pieces.

Reply:
Something like 
```
[RVR] Loading shared module: drowning
[RVR]     Including sv_drowning.lua
[RVR]     Adding clientside lua file cl_drowning.lua
```
?

Reply:
a print for every file? that would be alot of prints

Reply:
Could also do
```
[RVR] Loading shared module: drowning
[RVR]     Server includes:
[RVR]        sv_drowning.lua, sv_otherfile.lua
[RVR]     Adding clientside lua files
[RVR]         cl_drowning.lua
```

Reply:
Yeah or something like 

```
[RVR] Loading module: drowning
    - Loading sv_drowning.lua
    - Loading cl_drowning.lua
```

How it looks doesn't really matter to me per se just so long as it displays the different modules separately.

Reply:
> a print for every file? that would be alot of prints

I do also have that concern, wondering if it could be compacted with the above design

Reply:
Listing the number of files in each realm still tells you if all your files have been included, but it much more compact

Reply:
How about we have some kind of debug command to see which files actually loaded?
Like for example: `rvr_loaded_files <module>`.

Reply:
And it will just show the module file loaded on startup
```
[RVR] Loaded module: drowning
```

Reply:
Example for file counts:
```
[RVR] Loading modules...
[RVR] Loaded shared module: drowning - SERVER: 1 | CLIENT: 2
[RVR] Loaded client module: hud - CLIENT: 1
[RVR] Loaded shared module: hunger - SERVER: 2 | CLIENT: 1
[RVR] Finished loading modules
```



> How about we have some kind of debug command to see which files actually loaded?
> Like for example: `rvr_loaded_files <module>`.

So we'd store lists of files loaded? We could use peri's command lib for this when its done.

Reply:
Should I just leave this how it is? Then we can add the command once peri's module is merged?

Reply:
Yeah, sure let's send it

Reply:
Merge away, captain!